mod_ci/controllers.py

```diff
@@ -1110,8 +1240,14 @@ def progress_type_request(log, test, test_id, request) -> bool:
         log.error(f'Got an exception while posting to GitHub! Message: {a.message}')
 
     if status in [TestStatus.completed, TestStatus.canceled]:
-        # Start next test if necessary, on the same platform
-        start_platforms(g.db, repository, 60, test.platform)
+        # Delete the current instance
+        from run import config
+        compute = get_compute_service_object()
+        zone = config.get('ZONE', '')
+        project = config.get('PROJECT_NAME', '')
+        vm_name = f"{test.platform.value}-{test.id}"
+        operation = delete_instance(compute, project, zone, vm_name)
+        wait_for_operation(compute, project, zone, operation['name'])
```

Comment:
I'm thinking of a scenario here - what would happen when for some reason the operation to delete an instance fails? Would we be stuck in an infinite loop?

Reply:
No, the ```wait_for_operation``` function only waits till the API response is obtained, in case deletion of the VM instance fails. It simply moves forward, and the VM instance would be attempted to delete again just before the next cron job as per https://github.com/CCExtractor/sample-platform/pull/690/commits/04a705d2005aa97e41fb887d12cc9440d6fff306